sympy/physics/continuum_mechanics/beam.py

```diff
@@ -1825,41 +1848,72 @@ def solve_for_ild_reactions(self, value, *reactions):
             >>> E, I = symbols('E, I')
             >>> R_0, R_10 = symbols('R_0, R_10')
             >>> b = Beam(10, E, I)
-            >>> p0 = b.apply_support(0, 'roller')
+            >>> p0 = b.apply_support(0, 'pin')
             >>> p10 = b.apply_support(10, 'roller')
             >>> b.solve_for_ild_reactions(1,R_0,R_10)
             >>> b.ild_reactions
-            {R_0: x/10 - 1, R_10: -x/10}
+            {R_0: -SingularityFunction(a, 0, 0) + SingularityFunction(a, 0, 1)/10 - SingularityFunction(a, 10, 1)/10,
+            R_10: -SingularityFunction(a, 0, 1)/10 + SingularityFunction(a, 10, 0) + SingularityFunction(a, 10, 1)/10}
```

Comment:
`a` does not seem accessible to the user, as its creation is hidden in the Beam class.

Reply:
`a` does not have a value. It is a variable like `x` in this case. It describes the locaiton of the moving force of the beam. 

It works the same as the `x` in normal equations. The bending moment of a beam can be a function that is dependend of `x`. Simple example: 
`b.bending_moment()`
output: 10 - x 

`x = Symbol('x')`
`b.bending_moment().subs(x, 2)`
output: 8

This works the same way for the influence lines. But now the location of the moving force is the variable. 
`b.ild_reactions[r0]`
output: -SingularityFunction(a, 0, 0) + SingularityFunction(a, 0, 1)/10 - SingularityFunction(a, 10, 1)/10 

`a = Symbol('a')`
`b.ild_reactions[r0].subs(a, 1)`
output: -9/10

So I think `a` does not have to be accesible for the user. It works the same way as `x` in not ILD calculations.

Reply:
It is only chance that `a = Symbol('a')` is the same (as in `a is a_inside_Beam`) and there is no way for the user to know that it isn't `a =Symbol('a', positive=True)` or `a  = Dummy('a')`. The only way to access the guaranteed-to-be `a` is to do something like:

```python
In [1]: import sympy as sm

In [2]: x = sm.symbols('x')

In [3]: s = sm.SingularityFunction(x, 3, 2)

In [4]: s
Out[4]: SingularityFunction(x, 3, 2)

In [5]: s.args
Out[5]: (x, 3, 2)

In [6]: s.args[0]
Out[6]: x

In [7]: s = sm.SingularityFunction(x, 3, 2) + sm.SingularityFunction(x, 2, 4)

In [8]: s.args
Out[8]: (SingularityFunction(x, 2, 4), SingularityFunction(x, 3, 2))

In [9]: s.args[0].args
Out[9]: (x, 2, 4)

In [10]: s.args[0].args[0]
Out[10]: x
```

At the minimum a user should be able to pass in the variables to avoid creating them inside classes. For example:

```python
>>> def my_method(internal_symbol=sm.Symbol('a')):
          use(internal_symbol)
>>> a = Symbol('a', real=True)
>>> my_method(internal_symbol=a)
SingularityFunction(a, 2, 3)
>>> assert a is _.args[0]  # True!
```

Reply:
I see that for `x` it works as follows. 
It is set as a keyword argument in the `__init__` method, with `variable = Symbol('x')`
`self.variable = variable`, so `self.variable` is `x`. 

Then in each method it starts with `x = self.variable`. 
This way the user can get the variable by using `x = b.variable`.

Would it be a solution to do something similar with `a`?
So adding a keyword argument `ild_variable = Symbol('a')` and specifying `self.ild_variable = ild_variable`. 
This way the user can acces `a` by using `a = b.ild_variable`.

Reply:
Yes, that is preferable.

Reply:
I will do this

Reply:
Ok, thanks.